# DP: Link using --hash-style=both (alpha, amd64, armel, armhf, ia64, i386, powerpc, ppc64, s390, sparc)

2013-09-02 Davide Bettio <davide.bettio@ispirata.com>
        * Update patch to gcc 4.8.1

2012-01-10  Carsten Munk <carsten.munk@gmail.com> 
        * Switch to --hash-style=gnu

2006-07-11  Jakub Jelinek  <jakub@redhat.com>

	* config/i386/linux.h (LINK_SPEC): Add --hash-style=both.
	* config/i386/linux64.h (LINK_SPEC): Likewise.
	* config/rs6000/sysv4.h (LINK_OS_LINUX_SPEC): Likewise.
	* config/rs6000/linux64.h (LINK_OS_LINUX_SPEC32,
	LINK_OS_LINUX_SPEC64): Likewise.
	* config/s390/linux.h (LINK_SPEC): Likewise.
	* config/ia64/linux.h (LINK_SPEC): Likewise.
	* config/sparc/linux.h (LINK_SPEC): Likewise.
	* config/sparc/linux64.h (LINK_SPEC, LINK_ARCH32_SPEC,
	LINK_ARCH64_SPEC): Likewise.
	* config/alpha/linux-elf.h (LINK_SPEC): Likewise.

2009-12-21  Matthias Klose  <doko@ubuntu.com>

	* config/arm/linux-elf.h (LINK_SPEC): Add --hash-style=both.
--- a/gcc/config/alpha/linux-elf.h	2013-01-10 21:38:27.000000000 +0100
+++ b/gcc/config/alpha/linux-elf.h	2013-09-02 14:05:38.481130028 +0200
@@ -37,7 +37,7 @@
 
 #define ELF_DYNAMIC_LINKER	GNU_USER_DYNAMIC_LINKER
 
-#define LINK_SPEC "-m elf64alpha %{G*} %{relax:-relax}		\
+#define LINK_SPEC "-m elf64alpha --hash-style=gnu %{G*} %{relax:-relax}		\
   %{O*:-O3} %{!O*:-O1}						\
   %{shared:-shared}						\
   %{!shared:							\
--- a/gcc/config/i386/gnu-user.h	2013-02-08 19:08:12.000000000 +0100
+++ b/gcc/config/i386/gnu-user.h	2013-09-02 14:05:09.624986944 +0200
@@ -74,7 +74,7 @@
   { "link_emulation", GNU_USER_LINK_EMULATION },\
   { "dynamic_linker", GNU_USER_DYNAMIC_LINKER }
 
-#define GNU_USER_TARGET_LINK_SPEC "-m %(link_emulation) %{shared:-shared} \
+#define GNU_USER_TARGET_LINK_SPEC "-m %(link_emulation) --hash-style=gnu %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
--- a/gcc/config/i386/gnu-user64.h	2013-02-08 19:08:12.000000000 +0100
+++ b/gcc/config/i386/gnu-user64.h	2013-09-02 14:23:58.610585281 +0200
@@ -56,6 +56,7 @@
                   "%{" SPEC_64 ":-m " GNU_USER_LINK_EMULATION64 "} \
                    %{" SPEC_32 ":-m " GNU_USER_LINK_EMULATION32 "} \
                    %{" SPEC_X32 ":-m " GNU_USER_LINK_EMULATIONX32 "} \
+  --hash-style=gnu \
   %{shared:-shared} \
   %{!shared: \
     %{!static: \
--- a/gcc/config/ia64/linux.h	2013-01-10 21:38:27.000000000 +0100
+++ b/gcc/config/ia64/linux.h	2013-09-02 14:25:17.938978649 +0200
@@ -58,7 +58,7 @@
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-ia64.so.2"
 
 #undef LINK_SPEC
-#define LINK_SPEC "\
+#define LINK_SPEC " --hash-style=gnu \
   %{shared:-shared} \
   %{!shared: \
     %{!static: \
--- a/gcc/config/rs6000/linux64.h	2013-07-03 12:08:13.000000000 +0200
+++ b/gcc/config/rs6000/linux64.h	2013-09-02 14:00:33.787619139 +0200
@@ -385,11 +385,11 @@
 					   " -m elf64ppc")
 #endif
 
-#define LINK_OS_LINUX_SPEC32 LINK_OS_LINUX_EMUL32 " %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC32 LINK_OS_LINUX_EMUL32 " --hash-style=gnu %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " GNU_USER_DYNAMIC_LINKER32 "}}"
 
-#define LINK_OS_LINUX_SPEC64 LINK_OS_LINUX_EMUL64 " %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC64 LINK_OS_LINUX_EMUL64 " --hash-style=gnu %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " GNU_USER_DYNAMIC_LINKER64 "}}"
 
--- a/gcc/config/rs6000/sysv4.h	2013-07-03 12:08:13.000000000 +0200
+++ b/gcc/config/rs6000/sysv4.h	2013-09-02 13:50:50.372726141 +0200
@@ -788,7 +788,7 @@
 #define GNU_USER_DYNAMIC_LINKER \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER, UCLIBC_DYNAMIC_LINKER)
 
-#define LINK_OS_LINUX_SPEC "-m elf32ppclinux %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC "-m elf32ppclinux --hash-style=gnu %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " GNU_USER_DYNAMIC_LINKER "}}"
 
--- a/gcc/config/s390/linux.h	2013-01-10 21:38:27.000000000 +0100
+++ b/gcc/config/s390/linux.h	2013-09-02 13:49:49.288423250 +0200
@@ -65,7 +65,7 @@
 
 #undef  LINK_SPEC
 #define LINK_SPEC \
-  "%{m31:-m elf_s390}%{m64:-m elf64_s390} \
+  "%{m31:-m elf_s390}%{m64:-m elf64_s390} --hash-style=gnu \
    %{shared:-shared} \
    %{!shared: \
       %{static:-static} \
--- a/gcc/config/sparc/linux.h	2013-01-10 21:38:27.000000000 +0100
+++ b/gcc/config/sparc/linux.h	2013-09-02 13:47:29.063727902 +0200
@@ -86,7 +86,7 @@
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
 
 #undef  LINK_SPEC
-#define LINK_SPEC "-m elf32_sparc %{shared:-shared} \
+#define LINK_SPEC "-m elf32_sparc --hash-style=gnu %{shared:-shared} \
   %{!mno-relax:%{!r:-relax}} \
   %{!shared: \
     %{!static: \
--- a/gcc/config/arm/linux-elf.h	2013-01-10 21:38:27.000000000 +0100
+++ b/gcc/config/arm/linux-elf.h	2013-09-02 13:45:54.179257398 +0200
@@ -67,6 +67,7 @@
      %{rdynamic:-export-dynamic} \
      -dynamic-linker " GNU_USER_DYNAMIC_LINKER "} \
    -X \
+   --hash-style=gnu \
    %{mbig-endian:-EB} %{mlittle-endian:-EL}" \
    SUBTARGET_EXTRA_LINK_SPEC
 
